src/validation.cairo

```diff
@@ -49,9 +56,18 @@ impl TransactionValidatorImpl of TransactionValidator {
     }
 }
 
-fn block_hash(self: @ChainState, block: @Block, merkle_root: u256) -> Result<u256, ByteArray> {
-    // TODO: implement
-    Result::Ok(0)
+fn block_hash(
+    version: u32, previous_block: [u32; 8], merkle_root: [u32; 8], time: u32, bits: u32, nonce: u32
+    ) -> [
+    u32
+; 8] {
+    let mut header_data = ArrayTrait::<u32>::new();
+    header_data.append(version);
+    header_data.append(time);
+    header_data.append(bits);
+    header_data.append(nonce);
+
+    previous_block
```

Comment:
Now you should call `compute_sha256_u32_array` twice:
`compute_sha256_u32_array(compute_sha256_u32_array(data, 0, 0).span().into(), 0, 0)`